src/libs/verification/issues.ts

```diff
@@ -0,0 +1,122 @@
+import { IElement } from '@types';
+import {
+  VerificationAlternative,
+  VerificationAlternativeCode,
+} from './alternatives';
+import pluralize from 'pluralize';
+import { ClassStereotype } from '@constants/.';
+
+export enum IssueSeverity {
+  ERROR = 'ERROR',
+  WARNING = 'WARNING',
+}
+
+export enum VerificationIssueCode {
+  CLASS_NOT_UNIQUE_STEREOTYPE = 'CLASS_NOT_UNIQUE_STEREOTYPE',
+  CLASS_PLURAL_NAME = 'CLASS_PLURAL_NAME',
+  CLASS_INVALID_ONTOUML_STEREOTYPE = 'CLASS_INVALID_ONTOUML_STEREOTYPE',
+  CLASS_NON_ENUMERATION_WITH_LITERALS = 'CLASS_NON_ENUMERATION_WITH_LITERALS',
+  CLASS_ENUMERATION_WITH_PROPERTIES = 'CLASS_ENUMERATION_WITH_PROPERTIES',
+}
+
+/**
+ * Utility class for verification issues
+ *
+ * @author Claudenir Fonseca
+ * @author Lucas Bassetti
+ */
+export class VerificationIssue {
+  code: VerificationIssueCode;
+  title: string;
+  description: string;
+  source: IElement;
+  context: IElement[] | null;
+  severity: IssueSeverity;
+  alternatives: VerificationAlternative[] | null;
+
+  constructor(
+    code: VerificationIssueCode,
+    source: IElement,
+    context?: IElement[],
+  ) {
+    this.code = code;
+    this.title = null;
+    this.description = null;
+    this.source = source;
+    this.context = context ? context : null;
+    this.severity = null;
+    this.alternatives = null;
+
+    switch (code) {
```

Comment:
What do think about use object instead of switch? Like:

```
   const errors = {
      default: {
        title: `Unimplemented verification issue code: ${code}.`,
        ...
      },
      [VerificationIssueCode.CLASS_NOT_UNIQUE_STEREOTYPE]: {
        title: 'Not unique class stereotype.',
        description: `The class ${source.name} must have a unique OntoUML stereotype.`,
        severity: IssueSeverity.ERROR,
        alternatives: [
          new VerificationAlternative(
            VerificationAlternativeCode.REPLACE_ONTOUML_CLASS_STEREOTYPE,
            source,
          )
        ]
      },
      ...
    }

    const error = errors[code] || errors.default;

    this.code = code;
    this.title = error.title;
    this.description = error.description;
    this.source = source;
    this.context = context ? context : null;
    this.severity = error.severity;
    this.alternatives = error.alternatives;
```

Reply:
I don't know how is the memory allocation for switch vs object. Maybe we can do a research about that.